hazelcast/src/main/java/com/hazelcast/core/Cluster.java

```diff
@@ -169,6 +170,31 @@
     void changeClusterState(ClusterState newState, TransactionOptions transactionOptions);
 
     /**
+     * The cluster version indicates the operating version of the cluster. It is separate from each node's codebase version,
+     * as it may be acceptable for a node to operate at a different compatibility version than its codebase version. This method
+     * may return {@code null} if invoked after the ClusterService is constructed and before the node forms a cluster, either
+     * by joining existing members or becoming master of its standalone cluster if it is the first node on the cluster.
```

Comment:
Since the serialization service registers its own cluster version listener, it will never receive a `null` cluster version; when `Cluster.getClusterVersion()==null` it will initialize the listener with the node's codebase version.